homeassistant/components/sensibo/climate.py

```diff
@@ -232,10 +232,9 @@ def hvac_mode(self) -> HVACMode:
     @property
     def hvac_modes(self) -> list[HVACMode]:
         """Return the list of available hvac operation modes."""
-        if TYPE_CHECKING:
-            assert self.device_data.hvac_modes
-        hvac_modes = [SENSIBO_TO_HA[mode] for mode in self.device_data.hvac_modes]
-        return hvac_modes if hvac_modes else [HVACMode.OFF]
+        if not self.device_data.hvac_modes:
+            return [HVACMode.OFF]
```

Comment:
I'm a little confused by this new check.

- if `self.device_data.hvac_modes` can be `None` => is it a bug fix (the old code would have errored)?
- if `self.device_data.hvac_modes` can never be `None` => this line shouldn't be needed

Reply:
I just moved it around. It's the producing the same result as before.
It can't be none so I should actually in a follow-up fix that in the lib so the typing is correct. Could also replace with just type checking but that would change the behavior which is not for this PR.